src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs

```diff
@@ -117,8 +114,14 @@ public static string Concat(params object?[] args)
                 Debug.Assert(s != null);
                 Debug.Assert(position <= totalLength - s.Length, "We didn't allocate enough space for the result string!");
 
-                FillStringChecked(result, position, s);
+                CopyStringContent(result, position, s);
                 position += s.Length;
+
+                // Check for overflow
+                if (position < 0)
+                {
+                    ThrowHelper.ThrowOutOfMemoryException_StringTooLong();
```

Comment:
This check should not be needed here. Instead, `throw new OutOfMemoryException` in the previous look should use `ThrowOutOfMemoryException_StringTooLong`.